### PR TITLE
Fix issue and small changes

### LINKED ISSUE #10: User not added to group if added from Group repository
using function [`addMembers()`](https://github.com/Shubham20015/Sulaah/blame/b4eecf42fb5f3e10a3f8f9cf00850c332fd66c7e/src/main/java/com/expense/Sulaah/service/GroupService.java#L11) user is not added to Group  when saved with groupRepository.

### PR DESCRIPTION
**Pull request by Shubham20015**
Resolved #10 issue but look in @JsonIgnore

### CODE REVIEW

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java [hunk 1]
```diff
@@ -22,57 +22,59 @@ public class GroupServiceImpl implements GroupService {
...
-            if (usersInGroupList == null) {
-                usersInGroupList = new ArrayList<User>();
-                group.setUsersInGroup(usersInGroupList);
-            }
-            usersInGroupList.add(user.get());
-            return groupRepository.save(group);
-        } else {
-            return null;
+        if (!user.isPresent()) {
+			throw new SecurityException("User not present with given id: " + userId);
```
Comment (MASTREX) [inline src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java:29]: why security exception!

File: src/main/java/com/expense/Sulaah/controller/TransactionController.java [hunk 2]
```diff
@@ -29,9 +29,9 @@ private ResponseEntity<?> getTransactionsInGroup(@PathVariable int id) {
 	}
 
 	@PostMapping("/")
-	private Transaction addTransaction(@RequestBody TransactionDto transaction) {
-		return transactionService.addTransaction(transaction);
-
+	private Transaction addTransaction(@RequestBody TransactionDto transactionDto) {
+//		groupService.addMembers(transaction.getGroupId(),transaction.getUserIdWithShare().keySet().stream().toList());
```
Comment (MASTREX) [inline src/main/java/com/expense/Sulaah/controller/TransactionController.java]: remove this comment.
adding members from a transaction is not logically correct

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java [hunk 3]
```diff
@@ -22,57 +22,59 @@ public class GroupServiceImpl implements GroupService {
...
+		}
+
+        return group.get();
     }
 
     @Override
-    public String addMembers(int groupId, List<Integer> userIdList) {
-        Optional<Group> currentGroup = groupRepository.findById(groupId);
+    public Group addMembers(int groupId, List<Integer> userIdList) throws RuntimeException {
+        Optional<Group> groupDB = groupRepository.findById(groupId);
```
Comment (MASTREX) [inline src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java:54]: Naming!
you changed to 'currentGroup' in last commit :)

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java [hunk 4]
```diff
@@ -22,57 +22,59 @@ public class GroupServiceImpl implements GroupService {
     @Autowired
     private UserRepository userRepository;
 
+	private Group group;
```
Comment (MASTREX) [inline src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java]: This is pure evil

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java
Comment (MASTREX) [inline src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java:1]: fix this file in a new commit, remove 'group' class variable and use local variable in each method.

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -6,16 +6,21 @@ spring.output.ansi.enabled=ALWAYS
...
+#spring.datasource.dbname=app_fake_data
+
 spring.datasource.url=jdbc:mysql://localhost:3306/sulaahDB
 spring.datasource.username=root
 spring.datasource.password=samplePassword1
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
-spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
-# spring.datasource.initialization-mode=never
+
+#spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
```
Comment (MASTREX) [inline src/main/resources/application.properties]: I dont think this line should be commented

Comment (MASTREX) [reply src/main/resources/application.properties]: `spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect`
this one

**Review by MASTREX — CHANGES_REQUESTED** [review]
Fix ALL!

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java (hunk 1 above)
Comment (Shubham20015) [reply src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java:29]: If user is not present, then how it creates group so it is security concern

File: src/main/resources/application.properties (hunk 5 above)
Comment (Shubham20015) [reply src/main/resources/application.properties]: Uncomment it

File: src/main/java/com/expense/Sulaah/controller/TransactionController.java (hunk 2 above)
Comment (Shubham20015) [reply src/main/java/com/expense/Sulaah/controller/TransactionController.java]: Yeah, changed in a new commit

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java
Comment (Shubham20015) [reply src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java:1]: Fix errors in commit [2f180f0](https://github.com/Shubham20015/Sulaah/pull/13/commits/2f180f0b4d690278fac258b07a06a9dae61e3b6e)

File: src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java (hunk 3 above)
Comment (Shubham20015) [reply src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java:54]: Resolved

**Review by MASTREX — APPROVED** [review]

**Pull request merged**